### DISCUSSION TITLE
Should I use createRoutesStub when using React Router as a library?

### DISCUSSION THREAD

**Discussion by ArthurGuez** [remix-run/react-router#12576] [Q&A] [answered]
Hello there,

I'm trying to update React Router to v7 and I've noticed that the only doc page regarding testing is in the "Framework" section. It suggests to use `createRoutesStub` when your component depends on things like `useLoaderData` or `<Link>`. 

As of now, I was using that little abstraction to test such components:

```
export function renderWithRouter(routes: RouteObject[]) {
  const router = createMemoryRouter(routes);

  return render(<RouterProvider router={router} />);
}

it('should render header and footer', async () => {
  renderWithRouter([
    {
      element: <Layout />,
      loader: () => MOCK_LAYOUT_LOADER_DATA,
      path: '/',
    },
  ]);

  expect(await screen.findByTestId('header')).toBeTruthy();

  expect(await screen.findByTestId('footer')).toBeTruthy();
});
```

I plan to use React Router v7 as a library for now, does that mean there is no need to update my tests with `createRoutesStub`?

Thanks,

**Comment by sergiodxa** [comment, marked as answer]
Yes, the helper will helps you test components using hooks and components from RR without mocking them as they depend on RR context.

**Reply by ArthurGuez** [reply]
Great, thank you very much 🙏
